src/vulkan/engine_vulkan.cc

```diff
@@ -44,7 +45,7 @@ VkShaderStageFlagBits ToVkShaderStage(ShaderType type) {
       return VK_SHADER_STAGE_COMPUTE_BIT;
   }
 
-  // Unreachable
+  assert(false && "Vulkan::Unknown shader stage");
```

Comment:
asserts only fire on debug builds. Should we have a project-wide macro or function to express the unreachable condition even in release builds?
Question for @dj2 as well.

Reply:
There is an open bug to remove the asserts and return Results. Just have to do the refactoring.